### ISSUE TITLE
malformed_sstable_exception during bootsrap of new node

### ISSUE THREAD

**Issue by bentsi** [scylladb/scylla#4906]
*Installation details*
Scylla version (or git commit hash): 3.1.0.rc3-0.20190820.417250607
Cluster size: 4
OS (RHEL/CentOS/Ubuntu/AWS AMI): ami-0d86bdb3cdaf26671

Something bad happend to streamed sstables during bootstrap of new node 172.30.0.200 (after decommission)

Some of the errors:
- Corrupted range tombstone: invalid boundary type ?
- consumer not at partition boundary 
- Column value missing in current schema in sstable 

See details in logs. 
```
2019-08-23 08:13:27.810 | (DatabaseLogEvent Severity.CRITICAL): type=DATABASE_ERROR regex=Exception  line_number=2847 node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-7 [54.87.244.65 \| 172.30.0.200] (seed: False) 2019-08-23T08:13:16+00:00  ip-172-30-0-200 !ERR     \| scylla: [shard 4] compaction_manager - compaction failed: sstables::malformed_sstable_exception (Corrupted range tombstone: invalid boundary type ? in sstable /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-60-big-Data.db): retrying | DatabaseLogEvent, CRITICAL, events, DATABASE_ERROR
-- | -- | --
2019-08-23 08:13:27.811 | (DatabaseLogEvent Severity.CRITICAL): type=DATABASE_ERROR regex=Exception  line_number=2853 node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-7 [54.87.244.65 \| 172.30.0.200] (seed: False) 2019-08-23T08:13:18+00:00  ip-172-30-0-200 !ERR     \| scylla: [shard 5] compaction_manager - compaction failed: sstables::malformed_sstable_exception (Column value missing in current schema in sstable /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-117-big-Data.db): retrying | DatabaseLogEvent, CRITICAL, events, DATABASE_ERROR
2019-08-23 08:13:42.875 | (DatabaseLogEvent Severity.CRITICAL): type=DATABASE_ERROR regex=Exception  line_number=2894 node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-7 [54.87.244.65 \| 172.30.0.200] (seed: False) 2019-08-23T08:13:28+00:00  ip-172-30-0-200 !ERR     \| scylla: [shard 1] compaction_manager - compaction failed: sstables::malformed_sstable_exception (consumer not at partition boundary in sstable /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-57-big-Data.db): retrying | DatabaseLogEvent, CRITICAL, events, DATABASE_ERROR
2019-08-23 08:13:57.903 | (DatabaseLogEvent Severity.CRITICAL): type=DATABASE_ERROR regex=Exception  line_number=2940 node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-7 [54.87.244.65 \| 172.30.0.200] (seed: False) 2019-08-23T08:13:43+00:00  ip-172-30-0-200 !ERR     \| scylla: [shard 5] compaction_manager - compaction failed: sstables::malformed_sstable_exception (Column value missing in current schema in sstable /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-117-big-Data.db): retrying | DatabaseLogEvent, CRITICAL, events, DATABASE_ERROR
2019-08-23 08:13:57.908 | (DatabaseLogEvent Severity.CRITICAL): type=DATABASE_ERROR regex=Exception  line_number=2960 node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-7 [54.87.244.65 \| 172.30.0.200] (seed: False) 2019-08-23T08:13:51+00:00  ip-172-30-0-200 !ERR     \| scylla: [shard 11] compaction_manager - compaction failed: sstables::malformed_sstable_exception (Corrupted range tombstone: invalid boundary type #015 in sstable /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-333-big-Data.db): retrying | DatabaseLogEvent, CRITICAL, events, DATABASE_ERROR
2019-08-23 08:14:12.940 | (DatabaseLogEvent Severity.CRITICAL): type=DATABASE_ERROR regex=Exception  line_number=2980 node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-7 [54.87.244.65 \| 172.30.0.200] (seed: False) 2019-08-23T08:14:02+00:00  ip-172-30-0-200 !ERR     \| scylla: [shard 13] compaction_manager - compaction failed: sstables::malformed_sstable_exception (consumer not at partition boundary in sstable /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-223-big-Data.db): retrying | DatabaseLogEvent, CRITICAL, events, DATABASE_ERROR
2019-08-23 08:14:12.946 | (DatabaseLogEvent Severity.CRITICAL): type=DATABASE_ERROR regex=Exception  line_number=3002 node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-7 [54.87.244.65 \| 172.30.0.200] (seed: False) 2019-08-23T08:14:10+00:00  ip-172-30-0-200 !ERR     \| scylla: [shard 8] compaction_manager - compaction failed: sstables::malformed_sstable_exception (Corrupted range tombstone: invalid boundary type ???n sstable /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-428-big-Data.db): retrying | DatabaseLogEvent, CRITICAL, events, DATABASE_ERROR
2019-08-23 08:14:12.946 | (DatabaseLogEvent Severity.CRITICAL): type=DATABASE_ERROR regex=Exception  line_number=3005 node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-7 [54.87.244.65 \| 172.30.0.200] (seed: False) 2019-08-23T08:14:11+00:00  ip-172-30-0-200 !ERR     \| scylla: [shard 6] compaction_manager - compaction failed: sstables::malformed_sstable_exception (Corrupted range tombstone: invalid boundary type ??in sstable /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-62-big-Data.db): retrying | DatabaseLogEvent, CRITICAL, events, DATABASE_ERROR
2019-08-23 08:14:27.987 | (DatabaseLogEvent Severity.CRITICAL): type=DATABASE_ERROR regex=Exception  line_number=3008 node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-7 [54.87.244.65 \| 172.30.0.200] (seed: False) 2019-08-23T08:14:14+00:00  ip-172-30-0-200 !ERR     \| scylla: [shard 0] compaction_manager - compaction failed: sstables::malformed_sstable_exception (consumer not at partition boundary in sstable /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-840-big-Data.db): retrying | DatabaseLogEvent, CRITICAL, events, DATABASE_ERROR
2019-08-23 08:14:43.016 | (DatabaseLogEvent Severity.CRITICAL): type=DATABASE_ERROR regex=Exception  line_number=3037 node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-7 [54.87.244.65 \| 172.30.0.200] (seed: False) 2019-08-23T08:14:28+00:00  ip-172-30-0-200 !ERR     \| scylla: [shard 1] compaction_manager - compaction failed: sstables::malformed_sstable_exception (consumer not at partition boundary in sstable /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-57-big-Data.db): retrying | DatabaseLogEvent, CRITICAL, events, DATABASE_ERROR
```

db log
```
09/mc-50-big-Data.db:level=0, ]
2019-08-23T08:05:29+00:00  ip-172-30-0-200 !ERR     | scylla: [shard 12] compaction_manager - compaction failed: sstables::malformed_sstable_exception (Closing range to
mbstone that wasn't opened: clustering ckp{}, kind excl end, tombstone {tombstone: timestamp=1442880000000054, deletion_time=1442880057} in sstable /var/lib/scylla/data
/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-54-big-Data.db): retrying
2019-08-23T08:05:29+00:00  ip-172-30-0-200 !INFO    | scylla: [shard 12] compaction_manager - compaction task handler sleeping for 5 seconds
2019-08-23T08:05:29+00:00  ip-172-30-0-200 !INFO    | scylla: [shard 10] compaction - Compacted 4 sstables to [/var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9
adff000000000009/mc-108-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-122-big-Data.db:level=1, /var/lib/scylla/data/
keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-136-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-150-big-Da
ta.db:level=1, /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-164-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-3af16ce0c4
2711e9adff000000000009/mc-178-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-192-big-Data.db:level=1, ]. 1184537210 b
ytes to 1180635970 (~99% of original) in 11259ms = 100.00MB/s. ~1092096 total partitions merged to 1091905.
2019-08-23T08:05:30+00:00  ip-172-30-0-200 !INFO    | scylla: [shard 13] compaction - Compacting [/var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000
009/mc-55-big-Data.db:level=0, ]
2019-08-23T08:05:31+00:00  ip-172-30-0-200 !INFO    | scylla: [shard 8] stream_session - [Stream #a9775f20-c57c-11e9-b8e1-000000000005] Write to sstable for ks=keyspace
1, cf=standard1, estimated_partitions=3234862, received_partitions=854279
2019-08-23T08:05:31+00:00  ip-172-30-0-200 !INFO    | scylla: [shard 0] compaction - Compacting [/var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff0000000000
09/mc-28-big-Data.db:level=0, ]
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !WARNING | scylla: [shard 9] seastar_memory - oversized allocation: 16498688 bytes. This is non-fatal, but could lead to late
ncy and/or fragmentation issues. Please report: at   0x451d4bb#012  0x401dff3#012  0x402262b#012  0x14f45b7#012  0x151e04e#012  0x15215ba#012  0x15243b1#012  0x1581e32#
012  0x15a52f9#012  0x15a7ec4#012  0xd06a2c#012  0xd07a47#012  0x1094d8f#012  0x1097a3d#012  0x10987dd#012  0x109ba33#012  0x109d185#012  0x15e02d2#012  0x15bc4cf#012  0x837ec1
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !WARNING | scylla: [shard 9] large_data - Writing large cell keyspace1/standard1: 4c50304f3035374c3031C0 (16498446 bytes)
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: Segmentation fault on shard 9.
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: Backtrace:
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x0000000004180942
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x000000000406d9c5
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x000000000406dcc5
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x000000000406dd13
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x00007fdb27f7902f
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x00007fdb279a7a34
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x0000000001bcf115
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x000000000251851a
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x00000000014ac632
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x000000000406aeb1
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x000000000406b0ae
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x000000000414d0fd
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x000000000417773b
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: 0x000000000403916d
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: /opt/scylladb/libreloc/libpthread.so.0+0x000000000000858d
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | scylla: /opt/scylladb/libreloc/libc.so.6+0x00000000000fd6a2
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | kernel: reactor-9[2296]: segfault at 0 ip 00007fdb279a7a35 sp 00007fdb23de87b8 error 4 in libc.so.6[7fdb27868000+14d000]
2019-08-23T08:05:32+00:00  ip-172-30-0-200 !INFO    | kernel: Code: 00 00 0f 1f 00 31 c0 c5 f8 77 c3 66 2e 0f 1f 84 00 00 00 00 00 f3 0f 1e fa 89 f9 48 89 fa c5 f9 ef c0 83 e1 3f 83 f9 20 77 2b <c5> fd 74 0f c5 fd d7 c1 85 c0 0f 85 eb 00 00 00 48 83 c7 20 83 e1
2019-08-23T08:07:07+00:00  ip-172-30-0-200 !NOTICE  | systemd: scylla-server.service: main process exited, code=killed, status=11/SEGV
```
There are too many Seg faults, coredumps and tracebacks on this node.
The cluster and the node with corrupted sstables is alive.
Node with corrupted sstables: `54.87.244.65`

**Comment by avikivity** [comment]
```
-rw-r--r--. 1 scylla scylla     29900 Aug 23 08:05 /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-55-big-CRC.db
-rw-r--r--. 1 scylla scylla 489753880 Aug 23 08:05 /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-55-big-Data.db
-rw-r--r--. 1 scylla scylla        10 Aug 23 08:05 /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-55-big-Digest.crc32
-rw-r--r--. 1 scylla scylla   2313008 Aug 23 08:05 /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-55-big-Filter.db
-rw-r--r--. 1 scylla scylla   8031032 Aug 23 08:05 /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-55-big-Index.db
-rw-r--r--. 1 scylla scylla     61744 Aug 23 08:05 /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-55-big-Scylla.db
-rw-r--r--. 1 scylla scylla      4635 Aug 23 08:05 /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-55-big-Statistics.db
-rw-r--r--. 1 scylla scylla    269420 Aug 23 08:05 /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-55-big-Summary.db
-rw-r--r--. 1 scylla scylla        90 Aug 23 08:04 /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-55-big-TOC.txt
```

**Comment by avikivity** [comment]
That sstable was created 3 days ago.

**Comment by avikivity** [comment]
On shard 13

**Comment by avikivity** [comment]
Likely a duplicate of #4789. @asias please confirm.

**Comment by avikivity** [comment]
```
[root@ip-172-30-0-200 centos]# rpm -q scylla-server
scylla-server-3.1.0.rc3-0.20190820.417250607.el7.x86_64
```

**Comment by avikivity** [comment]
So it can't be #4789, because the fix for that is contained in 417250607.

**Comment by avikivity** [comment]
Please collect the logs around Aug 23 08:05:26 (10 minutes in each direction) from all nodes.

**Comment by bentsi** [comment]
all the logs are located on `3.82.110.145:/home/jenkins/slave/workspace/scylla-3.1/longevity/longevity-200gb-48h/scylla-cluster-tests/20190821-151304-500299/hosts`
I can cut them if it is needed

**Comment by avikivity** [comment]
@asias please look at it as well.

**Comment by avikivity** [comment]
Combined log file around the issue:

[sorted.txt](https://github.com/scylladb/scylla/files/3544725/sorted.txt)

**Comment by avikivity** [comment]
Starting the node node:

```
2019-08-23T08:02:32+00:00  ip-172-30-0-200 !INFO    | scylla_ami_setup: [INFO] --clustername longevity-200gb-48h-verify-limited--db-cluster-5b293a44 --totalnodes 4 --stop-services --seeds 172.30.0.189,172.30.0.243  --bootstrap true  --base64postscript=IyEvYmluL2Jhc2gKc3VkbyBlY2hvICdhY3Rpb24odHlwZT0ib21md2QiIFRhcmdldD0iMTcyLjMwLjAuMTczIiBQb3J0PSIzMjgwNyIgUHJvdG9jb2w9InRjcCIpJz4+IC9ldGMvcnN5c2xvZy5jb25mCnN1ZG8gc3lzdGVtY3RsIHJlc3RhcnQgcnN5c2xvZwoKc3VkbyBzZWQgLWkgJ3MvI01heFNlc3Npb25zIFwoLipcKSQvTWF4U2Vzc2lvbnMgMTAwMC8nIC9ldGMvc3NoL3NzaGRfY29uZmlnCnN1ZG8gc3lzdGVtY3RsIHJlc3RhcnQgc3NoZApzZWQgLWkgLWUgJ3MvXlwqW1s6Ymxhbms6XV0qc29mdFtbOmJsYW5rOl1dKm5wcm9jW1s6Ymxhbms6XV0qNDA5Ni8qCQlzb2Z0CW5wcm9jCQl1bmxpbWl0ZWQvJyAgICAgICAgICAgICAgICAvZXRjL3NlY3VyaXR5L2xpbWl0cy5kLzIwLW5wcm9jLmNvbmYKZWNobyAtZSAnKgkJaGFyZAlucHJvYwkJdW5saW1pdGVkJyA+PiAvZXRjL3NlY3VyaXR5L2xpbWl0cy5kLzIwLW5wcm9jLmNvbmYK
```

**Comment by avikivity** [comment]
```
2019-08-23T08:04:38+00:00  ip-172-30-0-200 !INFO    | scylla: [shard 0] storage_service - JOINING: Starting to bootstrap...
```

**Comment by avikivity** [comment]
We may have lost the important logs due to the compaction manager problem:

```
2019-08-23T08:04:45+00:00  ip-172-30-0-243 !INFO    | journal: Suppressed 1363 messages from /system.slice/scylla-server.service
```

**Comment by avikivity** [comment]
@raphaelsc the LCS compaction spinning looks like a serious bug and it's blocking investigation of a more serious bug.

**Comment by tgrabiec** [comment]
The corruption starts after the header for partition `000a31344c304d334d4f4b30`. The key, deletion time, and deletion timestamp look good. But the flags field (`0xcd`) which follows that looks wrong. It's interpreted as a partition end (`0x01`), but it's not really a partition end since no other bits should be set. 

What follows looks like total garbage to me.

In addition to that, according to the partition index, the next partition should start 1064 bytes later. But it doesn't seem to be there, there's only some garbage.

```
TRACE 2019-08-28 14:16:51,849 [shard 0] sstable - mp_row_consumer_m 0x600008aba058: setup_for_partition(pk{000a34304d4e323637503230})
TRACE 2019-08-28 14:16:51,849 [shard 0] sstable - mp_row_consumer_m 0x600008aba058: setup_for_partition(pk{000a31344c304d334d4f4b30})
TRACE 2019-08-28 14:16:51,850 [shard 0] sstable - mp_row_consumer_m 0x600008aba058: setup_for_partition(pk{20f7cf85a13e20e5d76432296b16db1cfc8633658369aaf62d73a760a5cedc1cc315086fb1660d130fe49bed5050d8c41978eb2bb314507f8c81539b59a8772e1eb6af0b9a0cb61568316bb65461c498266b8274de6fe746f7f65ab5b1568d35f55223a7bdff772849a3f1a2c5b01812084ad8a8028a5793bf44f4e965784d2dd5bae48d7e203e2e37a8ce6880758d33427cfa88f466e9c589aff435d328ff01fa260f24a2e6bec04e351ea231b6149bc55a13fc1a88d0e2c9766a9d80346ebab9e6f3e26c1c84a75b5595680e2e2b17d6c7a21393bf4fca937cbe89cd650023b60deadbae89246ecb13f19367463f8f57a578beeea9290ab74b8acd665fa08892e263e59ce60b821b3e04ccc8e7e84dc1d9f470dc92ed1ca
```

```
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a30344b4e4b3030364e30}, token:-8514760786125742531} -> 184140096
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a3734354c354c4b393031}, token:-8514760720643851727} -> 184141160
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a4c373236504b4c4c4c30}, token:-8514760648756841412} -> 184142224
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a35304e394c38324f3430}, token:-8514760513825428542} -> 184143288
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a33334c39324f4f343031}, token:-8514760508555624387} -> 184144352
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a32304b39343830383430}, token:-8514760479239089017} -> 184145416
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a4f4e3639383533383130}, token:-8514760393028492609} -> 184146480
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a503933394f4f334f3131}, token:-8514760383300136250} -> 184147544
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a35384f324b314b393531}, token:-8514760370203201925} -> 184148608
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a4f504d374d4d4e374e30}, token:-8514760297208663492} -> 184149672
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a39324c32323537343031}, token:-8514760258069395393} -> 184150736
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a34384c313737304f3830}, token:-8514760189699037032} -> 184151800
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a4f4f36324c4c33393631}, token:-8514760180627104779} -> 184152864
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a4e4e3038393630504f30}, token:-8514760129709228058} -> 184153928
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a4e364e324c3539334e30}, token:-8514760041030768063} -> 184154992
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a32314e324f33374e3631}, token:-8514759972306889165} -> 184156056
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a34304d4e323637503230}, token:-8514759965613837749} -> 184157120
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a31344c304d334d4f4b30}, token:-8514759829706055434} -> 184158184
TRACE 2019-08-28 15:12:11,616 [shard 0] sstable -   {key: pk{000a334f503239394e4e3330}, token:-8514759824064379506} -> 184159248
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - index 0x600001057400: old page index = 0
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - index 0x600001057400: new page index = 35, pos=184158184
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - index 0x600001057400: advance_upper_past({position: clustered,ckp{},1})
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - index 0x600001057400: no promoted index
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - index 0x6000010574b0 bound 0x600001057400: advance_to_next_partition()
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - parser 0x60000c050f00: setup_columns(): name=76616c7565 id=0
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - parser 0x60000c050f00: setup_columns(): name=4330 id=0
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - reader 0x600008faa400: reading partition
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - reader 0x600008faa400: read from index
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - reader 0x600008faa400: no tombstone
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - reader 0x600008faa400: read from data file
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - parser 0x60000c050f00: do_process_state(), off=184158184, state=PARTITION_START, buf=000a31344c304d334d4f4b307fffffff8000000000000000cd20f7cf85a13e20e5d76432296b16db1cfc8633658369aaf62d73a760a5cedc1cc315086fb1660d130fe49bed5050d8c41978eb2bb314507f8c81539b59a8772e1eb6af0b9a0cb61568316bb65461c498266b8274de6fe746f7f65ab5b1568d35f55223a7bdff772849a3f1a2c5b01812084ad8a8028a5793bf44f4e965784d2dd5bae48d7e203e2e37a8ce6880758d33427cfa88f466e9c589aff435d328ff01fa260f24a2e6bec04e351ea231b6149bc55a13fc1a88d0e2c9766a9d80346ebab9e6f3e26c1c84a75b5595680e2e2b17d6c7a21393bf4fca937cbe89cd650023b60deadbae89246ecb13f19367463f8f57a578beeea9290ab74b8acd665fa08892e263e59ce60b821b3e04ccc8e7e84dc1d9f470dc92ed1caf7e13bda23a1cc50b33ae8d4e4fbed494bb6a95d65db34f6deea9cdc0b62abbc0609abb460be99fcb47e3950b67260a774a7473e3fee2222893e2f8006e5145854eea9b2689f4c0439300465a236568419ca7d6ca5db90a55a5798b95883c74a3da7ba7912eb048cf7b6353003a848b9dffa053b0095657d8efd74fcf932b5025307b10b45058bcd510bf67c1fbcf5294d0002509d25916689e273944be8a566d59d671cbae6235ac554fa75fdc9326dbcea4dd798a660cb943299125220069a4731fa6cf864a0dd56a841d72958a4eb11d84938122dd6397af78f465e9956a87781b1e3571b51b4534d034353d75fa15c1952d477b87549ce3f152fb609d7f514695118ecbcec97b66fdefe591f3a4b58d0d0e64ab2aebd03a9073c1cfbb17c11d185812dded2e8397ba6825edb6d1d68ff123117d060472621ca122adb414fb59af14295a92d900952341837a79e3f82a550369c7f210bc922182c2f2762b7e303c17a0410dd7827642944932fba9879330d7dc84a1e3f14808bca64e1d8a98b56fc266d2f04cdc8536054d7d1c34aa05dd93b14623093f2ca2cdf2671b5a096bd5f3fc355f399ec8e1079172d8fe705192417e761fb014f521d778fb0d56e08c493408ff44195ec73468022fe26c9bf67d4d3541b0fa30e40801ecb9c7c9f946d259bf23f5b5a6bfe19035bbf64b6383bec8b15458a8816106f4d2a0417812d9f3f4233325b63d76c25b693a1e1fa75e2f501001b52a84fec86b792d1a9c8990456b8c3a71879657fefac0e999dbc135405b8fb2b717b1049dd95391708d48433053230eb3083acb409ca9a44d7c74e8291ab3519c6bf4b703716e5f00f33ff152691e9cc2a51c7aead37cc9a2fb49b4d0d67531bb36f67bcdb9f1e951d3a694c703eafb8d2a83cd0fea5f74569012a13767528bbfc499ce755f96f2f75523b37c45344cba7a89e6b60cf3f5f8b20d17a63b3eec572728b9bc61ab1771a913cf7ec4333164f8e8fe58eda1406fb44b970f4bdd1abe2bada5ad30d1aa65ab6e5b1775f2f9b38a3d30dab2b4fb5b2bda9b967d701a01
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - parser 0x60000c050f00: off=184158184, state=PARTITION_START
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - mp_row_consumer_m 0x600008faa458: consume_partition_start(deltime=(2147483647, -9223372036854775808)), _is_mutation_end=1
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - mp_row_consumer_m 0x600008faa458: setup_for_partition(pk{000a31344c304d334d4f4b30})
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - parser 0x60000c050f00: do_process_state(), off=184158208, state=FLAGS, buf=cd20f7cf85a13e20e5d76432296b16db1cfc8633658369aaf62d73a760a5cedc1cc315086fb1660d130fe49bed5050d8c41978eb2bb314507f8c81539b59a8772e1eb6af0b9a0cb61568316bb65461c498266b8274de6fe746f7f65ab5b1568d35f55223a7bdff772849a3f1a2c5b01812084ad8a8028a5793bf44f4e965784d2dd5bae48d7e203e2e37a8ce6880758d33427cfa88f466e9c589aff435d328ff01fa260f24a2e6bec04e351ea231b6149bc55a13fc1a88d0e2c9766a9d80346ebab9e6f3e26c1c84a75b5595680e2e2b17d6c7a21393bf4fca937cbe89cd650023b60deadbae89246ecb13f19367463f8f57a578beeea9290ab74b8acd665fa08892e263e59ce60b821b3e04ccc8e7e84dc1d9f470dc92ed1caf7e13bda23a1cc50b33ae8d4e4fbed494bb6a95d65db34f6deea9cdc0b62abbc0609abb460be99fcb47e3950b67260a774a7473e3fee2222893e2f8006e5145854eea9b2689f4c0439300465a236568419ca7d6ca5db90a55a5798b95883c74a3da7ba7912eb048cf7b6353003a848b9dffa053b0095657d8efd74fcf932b5025307b10b45058bcd510bf67c1fbcf5294d0002509d25916689e273944be8a566d59d671cbae6235ac554fa75fdc9326dbcea4dd798a660cb943299125220069a4731fa6cf864a0dd56a841d72958a4eb11d84938122dd6397af78f465e9956a87781b1e3571b51b4534d034353d75fa15c1952d477b87549ce3f152fb609d7f514695118ecbcec97b66fdefe591f3a4b58d0d0e64ab2aebd03a9073c1cfbb17c11d185812dded2e8397ba6825edb6d1d68ff123117d060472621ca122adb414fb59af14295a92d900952341837a79e3f82a550369c7f210bc922182c2f2762b7e303c17a0410dd7827642944932fba9879330d7dc84a1e3f14808bca64e1d8a98b56fc266d2f04cdc8536054d7d1c34aa05dd93b14623093f2ca2cdf2671b5a096bd5f3fc355f399ec8e1079172d8fe705192417e761fb014f521d778fb0d56e08c493408ff44195ec73468022fe26c9bf67d4d3541b0fa30e40801ecb9c7c9f946d259bf23f5b5a6bfe19035bbf64b6383bec8b15458a8816106f4d2a0417812d9f3f4233325b63d76c25b693a1e1fa75e2f501001b52a84fec86b792d1a9c8990456b8c3a71879657fefac0e999dbc135405b8fb2b717b1049dd95391708d48433053230eb3083acb409ca9a44d7c74e8291ab3519c6bf4b703716e5f00f33ff152691e9cc2a51c7aead37cc9a2fb49b4d0d67531bb36f67bcdb9f1e951d3a694c703eafb8d2a83cd0fea5f74569012a13767528bbfc499ce755f96f2f75523b37c45344cba7a89e6b60cf3f5f8b20d17a63b3eec572728b9bc61ab1771a913cf7ec4333164f8e8fe58eda1406fb44b970f4bdd1abe2bada5ad30d1aa65ab6e5b1775f2f9b38a3d30dab2b4fb5b2bda9b967d701a01
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - parser 0x60000c050f00: off=184158208, state=FLAGS
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - parser 0x60000c050f00: off=184158209, state=FLAGS_2, _flags=205
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - mp_row_consumer_m 0x600008faa458: consume_partition_end()
```

```
TRACE 2019-08-28 15:12:11,617 [shard 0] sstable - parser 0x60000c050f00: do_process_state(), off=184158184, state=PARTITION_START, buf=

000a31344c304d334d4f4b30 7fffffff 8000000000000000 cd20f7cf85a13e20e5d76432296b16db1cfc8633658369aaf62d73a760a5cedc1cc315086fb1660d130fe49bed5050d8c41978eb2bb314507f8c81539b59a8772e1eb6af0b9a0cb61568316bb65461c498266b8274de6fe746f7f65ab5b1568d35f55223a7bdff772849a3f1a2c5b01812084ad8a8028a5793bf44f4e965784d2dd5bae48d7e203e2e37a8ce6880758d33427cfa88f466e9c589aff435d328ff01fa260f24a2e6bec04e351ea231b6149bc55a13fc1a88d0e2c9766a9d80346ebab9e6f3e26c1c84a75b5595680e2e2b17d6c7a21393bf4fca937cbe89cd650023b60deadbae89246ecb13f19367463f8f57a578beeea9290ab74b8acd665fa08892e263e59ce60b821b3e04ccc8e7e84dc1d9f470dc92ed1caf7e13bda23a1cc50b33ae8d4e4fbed494bb6a95d65db34f6deea9cdc0b62abbc0609abb460be99fcb47e3950b67260a774a7473e3fee2222893e2f8006e5145854eea9b2689f4c0439300465a236568419ca7d6ca5db90a55a5798b95883c74a3da7ba7912eb048cf7b6353003a848b9dffa053b0095657d8efd74fcf932b5025307b10b45058bcd510bf67c1fbcf5294d0002509d25916689e273944be8a566d59d671cbae6235ac554fa75fdc9326dbcea4dd798a660cb943299125220069a4731fa6cf864a0dd56a841d72958a4eb11d84938122dd6397af78f465e9956a87781b1e3571b51b4534d034353d75fa15c1952d477b87549ce3f152fb609d7f514695118ecbcec97b66fdefe591f3a4b58d0d0e64ab2aebd03a9073c1cfbb17c11d185812dded2e8397ba6825edb6d1d68ff123117d060472621ca122adb414fb59af14295a92d900952341837a79e3f82a550369c7f210bc922182c2f2762b7e303c17a0410dd7827642944932fba9879330d7dc84a1e3f14808bca64e1d8a98b56fc266d2f04cdc8536054d7d1c34aa05dd93b14623093f2ca2cdf2671b5a096bd5f3fc355f399ec8e1079172d8fe705192417e761fb014f521d778fb0d56e08c493408ff44195ec73468022fe26c9bf67d4d3541b0fa30e40801ecb9c7c9f946d259bf23f5b5a6bfe19035bbf64b6383bec8b15458a8816106f4d2a0417812d9f3f4233325b63d76c25b693a1e1fa75e2f501001b52a84fec86b792d1a9c8990456b8c3a71879657fefac0e999dbc135405b8fb2b717b1049dd95391708d48433053230eb3083acb409ca9a44d7c74e8291ab3519c6bf4b703716e5f00f33ff152691e9cc2a51c7aead37cc9a2fb49b4d0d67531bb36f67bcdb9f1e951d3a694c703eafb8d2a83cd0fea5f74569012a13767528bbfc499ce755f96f2f75523b37c45344cba7a89e6b60cf3f5f8b20d17a63b3eec572728b9bc61ab1771a913cf7ec4333164f8e8fe58eda1406fb44b970f4bdd1abe2bada5ad30d1aa65ab6e5b1775f2f9b38a3d30dab2b4fb5b2bda9b967d701a01
```

**Comment by raphaelsc** [comment]
@avikivity I think we're incorrectly trying to demote a high-level sstable because parallel compaction is disabled for LCS and the strategy thinks we have gone many rounds without compacting high-levels. This bug could lead to increased write amplification. I'll open an issue

**Comment by raphaelsc** [comment]
opened issue for LCS issue: https://github.com/scylladb/scylla/issues/4919

**Comment by avikivity** [comment]
The corruption does not start on a buffer boundary, and does start at an object boundary. This points to a problem at the higher level of the stack and rules out filesystem or output_stream problems.

**Comment by avikivity** [comment]
I think it also rules out problems in streaming, since the flags field is generated by the sstable writer, not streaming.

**Comment by avikivity** [comment]
As a trivial way of estimating entropy, I compressed the garbage section and got slightly smaller or larger sizes. So it looks like already comressed or encrypted data.

**Comment by avikivity** [comment]
Does not disassemble into anything reasonable either.

**Comment by tgrabiec** [comment]
According to sstable writer code, there's no way a partition tombstone
would be written but incorrect flags were written if the file writers
behave correctly.

The flow is this:

    ensure_tombstone_is_written(); // This writes deletion time and
deletion timestamp, which looks ok in the buffer

    ...

    write(_sst.get_version(), *_data_writer, flags); // This should write
0xA0, we see 0xCD
    write(_sst.get_version(), *_data_writer,
row_extended_flags::is_static);  // This should write 0x01, we see 0x20

One explanation is that there is a use-after-free on the output stream and
it is used by another process concurrently. So the underlying buffer is
first filled with garbage by another process, then we write the partition
header, then the other process flushes the buffer.

The sstable writer seems oblivious to all this because it writes the
partition index for the next entry as if nothing happened. The next
partition entry falls in the garbage.

**Comment by avikivity** [comment]
reuse indeed explains the garbage (if it got reused by a compressed file writer, and maybe some system table sstables are being compacted now).

Reuse can happen if an exception happened during the write, but then we should not have attempted to read from the corrupted sstable.

**Comment by avikivity** [comment]
Maybe we should test bootstrap in debug mode.

**Comment by tgrabiec** [comment]
I analyzed a different sstable (generation 22, the oldest which is malformed) and the corruption there is more subtle.

The cell's value length seems to be encoded incorrectly. The MSB of it is 0x65, which means length of 0x65, instead of 0x84 followed by 0x00, which means length of 0x400 (=1024). 

If the value length is interpreted as a two-byte varint and we assume that it is 1024, then the buffer contents fit into the format, I land in the next partition entry:

```

a0 01 840b0000fc70996ee0eea8 65 4a

(the last two bytes above should be 84 00 to indicate length of 1024)

86ee58bbab4c706d6e8448006f4bb026602e895dc1a0a8c2acc2e32f1db4987e06a9baa9941c7a23d8ce0309a544b3692649bc8ce890628cce6117ea0362ecf070afcf4f03b52b5921e2cb33fc219cdf5609d8b96c86ac6f336c4726bc0dbd89ffcae00e9f28788f78cb475e57e746a5259ba39677924f4b730708308866b72dec7abdbba675b8a4781fbdf5de6d96ebb4d13d84a3e8cb11b988a77201c84aaf1b8f368bb507bd4c4918939169cf82f25c83bce9debc81bebaaf705042606041f6c7b3f3149d0c90cb6028c403143c2b31bdc09542f12524714758a658541d29b40a1a3656d86d46c141a3ea83df53fa36307099c67e7972cca6deb51eff01acb79b911cd5b5edabfaa9b6ed1b3b8272687eb82d2e185d72a6f33ab6573c0e175148fbd2318b1fbe9556e9524f623f8cad3b63bd95745ffe34496cbe98bdeb367c8be087518069aeaa02962b1ad34581dfeec789eb47623ffe0f7ed7dab8fdfc4c0912063e189d57093a2f179b59037b6fba6b396489591f950b2aa83f6611799cb9ccc6b28f373b72ef6a27dc882b583ec354ef08a98f49f7ba5dbc6f4761175e923467938c6a2bc09feeb5cfa87b16d86c145de61e0f6fbe3889f5bc85419517fec0b723ffe61c9b18087add03f98d2efc2da76901ddc4a723db58621962ca56722e9c2b75faf8085def4e01c220b1842c3380e09e0b6ed37c3e89b42acda4218edd3722071d43e756b4ca9d708b2714420d3a44a0e7edc3d6982fb5d3676ca6577bd5ea360f63073b73e3d47889bced79814d455ea4c2b4071249c40bcea69e2650fee28d200010140f0e2354e43031bf7a2727ea999c340c1304a06d7e6fb3314ae70b92faa8be934cc68f54d6b3eb001082d1c5e6c95b38cf6a8b60e6804a1cb2aef3b88964dc12f9b5bbd6b724235788aa6b3c7d5c97c6a13f8b8732e648726713c4bd0f333a283a462b2472577c7ca5d5b07a35fc1d4119b3b5d049469bba15d2e3c30d329668920fc4106330d7f629876b92d2a9543f80b916f48412877e465ad507e698a2a825d75d3dee607670fe3407076a36582f40efd43ee0c17d764a3dd6c671366827025af6cbed07bacc60e510343e4f436eefa26f06a62cfcc66b158dbdd79c53f3b10455d3371600f0ac9641077833eef7cb2e568ea44ee96025f199743994c029267c65f30058e2c67b92b7cf8f4934c39e2bbb8fe7c09e200f46023d113e089dd37476d501ea94230124fd19599309fa6c30efdf8bd7f7a339a76c2de0b096171625a4edcb256db05f3e680f0b67edb90244cc353b3c1928d4136fa80a779bd13d2ec021425e6ac79abc2add679a4b91283e3bc196cfe544b57623dd6bac820d002997665d49b12c708fa3adb1c7a3cfe5b6be40a53887b21fdf457bc76071049916e5a132408d3f9cfeb743f8ce5ea53bce9812680e

01 (partition end)

000a 4c4e504b363634383231 7fffffff 8000000000000000 a0 01 840b0000fc705791c5d4c6 84 
```

**Comment by tgrabiec** [comment]
It's actually a similar story with the first sstable. If we assume that the flags for the row and the cell are the same as in other partitions, we end up at the next partition.

And the partition index actually points not in the garbage, but where the partition actually starts. I must have made a mistake when calculating offsets the first time.

So it looks like something was writing in the buffer concurrently with the owning sstable writer.

**Comment by tgrabiec** [comment]
The "garbage" is just cell's value, which is random (cassandra stress workload)

**Comment by avikivity** [comment]
Aha. But c-s default workload is 5 60-byte columns, so I expected to see some punctuation there, and didn't.

**Comment by avikivity** [comment]
The concurrent writer could be the same sstable writer, but with a missing .get() causing it to run in parallel.

**Comment by avikivity** [comment]
Hard to see how that can happen. The output_stream is nicely encapsulated in file_writer. All future-returning calls have a matching .get().

**Comment by tgrabiec** [comment]
> Aha. But c-s default workload is 5 60-byte columns, so I expected to see some punctuation there, and didn't.

@bentsi Did you change the default size of the value in cassandra-stress?

It appears that only C0 is present in the sstable's schema, perhaps other columns were deleted? @bentsi Do you delete columns in this test?

**Comment by bentsi** [comment]
@tgrabiec 
1. yes, we use ` -col 'size=FIXED(1024) n=FIXED(1)' `
full commands that was used
```
"cassandra-stress write cl=ALL n=200200300  -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=1000 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..200200300 -log interval=15"

"cassandra-stress read cl=ONE duration=2860m -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=250  -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..200200300 -log interval=5"
```
2. AFAIK c-s doesn't delete columns, and we don't have such nemesis

**Comment by tgrabiec** [comment]
On Sun, Sep 1, 2019 at 10:04 AM Bentsi <notifications@github.com> wrote:

> @tgrabiec <https://github.com/tgrabiec>
>
>    1. yes, we use -col 'size=FIXED(1024) n=FIXED(1)'
>    full commands that was used
>
> "cassandra-stress write cl=ALL n=200200300  -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=1000 -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..200200300 -log interval=15"
>
> "cassandra-stress read cl=ONE duration=2860m -schema 'replication(factor=3) compaction(strategy=LeveledCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=250  -col 'size=FIXED(1024) n=FIXED(1)' -pop seq=1..200200300 -log interval=5"
>
>
>    1. AFAIK c-s doesn't delete columns, and we don't have such nemesis
>
>
n=FIXED(1) means there will be only one column set, so that explains we see
only C0.

**Comment by bentsi** [comment]
trying to reproduce on latest `3.1.0.rc5` with the exact sequence of Nemesis

**Comment by avikivity** [comment]
I suspected it was the lack of scylladb/seastar@7e2483ac867d7deff6fa6170c86667e1f5c147b0, however it is in fact present. It would have explained it by output_stream getting confused due to get yielding because it ran in the sstable_writer's thread when a yield was unexpected. However, that commit is in fact present. But maybe it's a related bug.

**Comment by avikivity** [comment]
It's even stranger. Cells are written to a bytes_ostream, not a file_writer, so they don't have any opportunity for a missed future::get().

**Comment by bentsi** [comment]
2 runs with exact sequence of Nemesis had not reproduced the issue... Running again

**Comment by bentsi** [comment]
Additional discoveries:
during this run one node was stuck for ~24 hours in decommision (probably related to #4971)
Last mesage in shutdown sequence was: Thrift server stopped. Then there were some compactions.
```
019-08-21T23:24:00+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] storage_service - DECOMMISSIONING: unbootstrap starts
2019-08-21T23:24:00+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] storage_service - LEAVING: replaying batch log and streaming data to other nodes
2019-08-21T23:24:00+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] range_streamer - Unbootstrap starts, nr_ranges_remaining=3183
2019-08-21T23:24:00+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] range_streamer - Unbootstrap with [172.30.0.189, 172.30.0.106, 172.30.0.243] for keyspace=keyspac
e1 started, nodes_to_stream=3
2019-08-21T23:24:00+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] range_streamer - Unbootstrap with 172.30.0.189 for keyspace=keyspace1, 24 out of 242 ranges: rang
es = 24
2019-08-21T23:24:00+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] stream_session - [Stream #c1768be0-c46a-11e9-b682-000000000005] Executing streaming plan for Unbo
...
2019-08-22T00:36:06+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 1] compaction - Compacting [/var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff00000000000
9/mc-112225-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-112239-big-Data.db:level=3, /var/lib/scylla/data/keyspace1
/standard1-3af16ce0c42711e9adff000000000009/mc-111441-big-Data.db:level=2, ]
2019-08-22T00:36:07+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] storage_service - DECOMMISSIONING: unbootstrap done
2019-08-22T00:36:07+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] storage_service - Thrift server stopped
2019-08-22T00:36:07+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 5] compaction - Compacted 9 sstables to [/var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9ad
```
the last message from the node before it stuck was
```
2019-08-22T00:37:11+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 1] compaction - Compacted 5 sstables to [/var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-114717-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-114731-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-114745-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-114759-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-3af16ce0c42711e9adff000000000009/mc-114773-big-Data.db:level=3, ]. 839911785 bytes to 763319241 (~90% of original) in 3376ms = 215.63MB/s. ~777600 total partitions merged to 706283.
```
during correct run the shutdown sequence is
```
2019-08-23T03:52:01+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] storage_service - DECOMMISSIONING: unbootstrap done
2019-08-23T03:52:01+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] storage_service - Thrift server stopped
2019-08-23T03:52:01+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] storage_service - CQL server stopped
2019-08-23T03:52:01+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] storage_service - DECOMMISSIONING: shutdown rpc and cql server done
2019-08-23T03:52:01+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] storage_service - DECOMMISSIONING: stop batchlog_manager done
2019-08-23T03:52:01+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] gossip - My status = LEFT
2019-08-23T03:52:01+00:00  ip-172-30-0-63 !WARNING | scylla: [shard 0] gossip - No local state or state is in silent shutdown, not announcing shutdown
2019-08-23T03:52:01+00:00  ip-172-30-0-63 !INFO    | scylla: [shard 0] storage_service - DECOMMISSIONING: stop_gossiping done
```
So it was stuck between `storage_service - Thrift server stopped` and stopping mechanism of the CQL server.

Currently running the same sequence of Nemesis with small dataset (1GB) and 5 min interval between Nemesis

**Comment by bhalevy** [comment]
According to @bentsi he wasn't able to reproduce this issue. Closing for now.
Please open if hit again.

**Comment by avikivity** [comment]
@bentsi did the toppartitions nemesis run on the node that caused the problems?

**Comment by bentsi** [comment]
1. toppartitions ran before the decommission nemesis that was stuck  on the same node (2019-08-21 23:23:21.087)
2. toppartitions ran before the last decommission that cause the issue but not on the same node
Exact sequence:
```
2019-08-21 15:58:19.886:  type=start name=MajorCompaction node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=None
2019-08-21 16:02:35.845:  type=end name=MajorCompaction node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=255
2019-08-21 16:35:31.946:  type=start name=SoftRebootNode node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-4 [3.87.221.222 | 172.30.0.132] (seed: False) duration=None
2019-08-21 16:38:47.968:  type=end name=SoftRebootNode node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-4 [3.87.221.222 | 172.30.0.132] (seed: False) duration=196
2019-08-21 17:11:49.034:  type=start name=MajorCompaction node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=None
2019-08-21 17:20:08.806:  type=end name=MajorCompaction node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=499
2019-08-21 17:53:11.047:  type=start name=TruncateMonkey node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=None
2019-08-21 17:53:50.449:  type=end name=TruncateMonkey node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=39
2019-08-21 18:26:50.903:  type=start name=StopStartService node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-4 [3.87.221.222 | 172.30.0.132] (seed: False) duration=None
2019-08-21 18:31:17.831:  type=end name=StopStartService node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-4 [3.87.221.222 | 172.30.0.132] (seed: False) duration=266
2019-08-21 19:04:18.244:  type=start name=Enospc node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=None
2019-08-21 19:07:17.174:  type=end name=Enospc node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=178
2019-08-21 19:40:23.013:  type=start name=Decommission node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-4 [3.87.221.222 | 172.30.0.132] (seed: False) duration=None
2019-08-21 22:16:15.645:  type=end name=Decommission node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-4 [3.87.221.222 | 172.30.0.132] (seed: False) duration=9352
2019-08-21 22:49:22.722:  type=start name=ShowTopPartitions node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=None
2019-08-21 22:50:24.185:  type=end name=ShowTopPartitions node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=61
2019-08-21 23:23:21.087:  type=start name=Decommission node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=None
2019-08-23 00:32:46.424:  type=start name=StopWaitStartService node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=None
2019-08-23 00:39:00.303:  type=end name=StopWaitStartService node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=373
2019-08-23 01:11:57.872:  type=start name=NodetoolCleanupMonkey node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-1 [3.86.82.27 | 172.30.0.189] (seed: True) duration=None
2019-08-23 01:23:53.797:  type=start name=NodetoolCleanupMonkey node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=None
2019-08-23 01:33:47.585:  type=start name=NodetoolCleanupMonkey node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-3 [3.87.218.182 | 172.30.0.243] (seed: True) duration=None
2019-08-23 01:45:15.391:  type=start name=NodetoolCleanupMonkey node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-5 [18.234.207.8 | 172.30.0.106] (seed: False) duration=None
2019-08-23 01:56:40.262:  type=end name=NodetoolCleanupMonkey node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=2685
2019-08-23 02:29:40.192:  type=start name=Decommission node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=None
2019-08-23 05:38:29.975:  type=end name=Decommission node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-2 [54.89.241.217 | 172.30.0.63] (seed: False) duration=11329
2019-08-23 06:11:38.940:  type=start name=ShowTopPartitions node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-5 [18.234.207.8 | 172.30.0.106] (seed: False) duration=None
2019-08-23 06:12:32.984:  type=end name=ShowTopPartitions node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-5 [18.234.207.8 | 172.30.0.106] (seed: False) duration=54
2019-08-23 06:45:49.810:  type=start name=Decommission node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-6 [3.82.204.241 | 172.30.0.210] (seed: False) duration=None
2019-08-23 12:40:04.083:  type=end name=Decommission node=Node longevity-200gb-48h-verify-limited--db-node-5b293a44-6 [3.82.204.241 | 172.30.0.210] (seed: False) duration=21254
```

**Comment by avikivity** [comment]
1. can explain why decomission got stuck (if early free happened to match some data structure used by tls)
2. I don't think it can explain the corruption then.
